### DISCUSSION TITLE
Help with install

### DISCUSSION THREAD

**Discussion by dukester** [modula3/cm3#784] [Q&A] [answered]
Old guy, wannabe cm3 user here. Just an amateur, hobbyist hacker with some Oberon2and Pascal experience.

I'm on a 64 bit laptop running Linux antiX distro. Dled a cm3 Linux release from http://www.opencm3.net/releng/download.html then stumbled on this site.

Would someone tell me what I need to clone to get going with your latest cm3 incarnation - please!

TIA and Merry Christmas to you all!
--
Duke
Alberta, Canada

**Comment by ghost** [comment, marked as answer]
Maybe this will help https://github.com/modula3/cm3/wiki/Getting-Started%3A-Linux

**Comment by dukester** [comment]
Thanks for the link. BTW, I’m more of an emacs/CLI person. Is there any chance of a “minimal” install with just compiler and libraries etc? Thanks again and Merry Christmas!
—
Duke

**Reply by ghost** [reply]
The default install is CLI only.

**Reply by dukester** [reply]
Got it installed! Zero probs! Would this mailing list also be the place to ask questions while learning the language? Or is there another more suitable venue for that? Thanks again!

**Reply by ghost** [reply]
I'm not sure what the best venue is.  Someone else may have to answer that.

We have:
- docs :: https://modula3.github.io/cm3/
- tutorial :: https://modula3.github.io/cm3/tutorial/m3/m3_toc.html
- a lot of good longer-form pdfs :: https://modula3.github.io/cm3/help/bib.html

I recommend the "Modula-3 Report" https://modula3.github.io/cm3/help/bib.html#m3-CDG*89

The book "Systems Programming with Modula-3" https://modula3.github.io/cm3/help/bib.html#SPwM3 is generally considered the best resource, but it is not available electronically that I'm aware of.

**Reply by MarcusE1W** [reply]
The book "Systems Programming with Modula-3" seems to be available as PDF here. No idea about the copy right situation.

https://archive.org/details/systems-programming-with-modula-3

**Reply by ghost** [reply]
:+1: That's a great find, I'll add it to our documentation page.

**Comment by dukester** [comment]
Great! Thx ….

—
Duke

**Comment by dukester** [comment]
On Sat, 25 Dec 2021 05:55:29 -0800
Eric Sessoms ***@***.***> wrote:

> I'm not sure what the best venue is.  Someone else may have to
> answer that.

OK! Got it ...

> We have:
> - docs :: https://modula3.github.io/cm3/
> - tutorial ::
> https://modula3.github.io/cm3/tutorial/m3/m3_toc.html
> - a lot of good longer-form pdfs ::
> https://modula3.github.io/cm3/help/bib.html
>
> I recommend the "Modula-3 Report"
> https://modula3.github.io/cm3/help/bib.html#m3-CDG*89
>
> The book "Systems Programming with Modula-3"
> https://modula3.github.io/cm3/help/bib.html#SPwM3 is generally
> considered the best resource, but it is not available
> electronically that I'm aware of.

Thanks for all that info!
--
Duke
** Text only please. Bottom post is best for me **

**Comment by ghost** [comment]
> Or is there another more suitable venue for that?

We have an open issue to setup a chat room https://github.com/modula3/cm3/issues/61  I'll see if I can get something in-place sooner rather than later.

**Comment by dukester** [comment]
On Sat, 01 Jan 2022 13:25:38 -0800
Eric Sessoms ***@***.***> wrote:

> > Or is there another more suitable venue for that?
>
> We have an open issue to setup a chat room
> https://github.com/modula3/cm3/issues/61  I'll see if I can get
> something in-place sooner rather than later.

If the word got out to the orphaned Modula2, Oberon, Oberon2 et al
folks that Modula-3 is alive and well and IS a close fit to what
they already know and probably miss to some extent, doing some
might just spark interest and commitment to Modula-3.

comp.lang.oberon still exists on Usenet. :)
--
Duke
** Text only please. Bottom post is best for me **

**Comment by mikanystrom** [comment]
Much of spwm3 (the Green Book) is available online. Language spec obviously
is. The threads chapter

https://www.hpl.hp.com/techreports/Compaq-DEC/SRC-RR-35.pdf

Trestle

https://www.hpl.hp.com/techreports/Compaq-DEC/SRC-RR-69.pdf

Network Objects

https://www.hpl.hp.com/techreports/Compaq-DEC/SRC-RR-115.pdf

I/O Streams

https://hplabs.itcs.hp.com/techreports/Compaq-DEC/SRC-RR-53.pdf

probably others I missed (HP's web site interface is terrible...)

Then there are a bunch of related papers, not strictly Modula-3 but closely
related:

https://hplabs.itcs.hp.com/techreports/Compaq-DEC/SRC-RR-64.pdf

+ other papers on Modula-2+

interesting rationale here

https://hplabs.itcs.hp.com/techreports/Compaq-DEC/SRC-RR-3.pdf

and there are some interesting papers on Obliq and Juno, and the Extended
Static Checker.  I'm sure others I have forgotten.

     Mika

**Comment by dukester** [comment]
On Sun, 02 Jan 2022 04:54:10 -0800
MarcusE1W ***@***.***> wrote:

> The book "Systems Programming with Modula-3" seems to be
> available as PDF here. No idea about the copy right situation.
>
> https://archive.org/details/systems-programming-with-modula-3

Thx for the link! I may never get to the point of being a system
hacker in _any_ language, never mind Modula-3. :)  If I can get a
handle on the basics than I might have a chance. LOL

I wish that Modula-3 would automagically get included at:
https://learnxinyminutes.com/

That would be a good start - would it not? Thanks again ...
--
Duke
** Text only please. Bottom post is best for me **
